pkg/requests/requests.go

```diff
@@ -0,0 +1,78 @@
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// ConstructGetURL creates a GET request from a base URL, a map of query
+// parameters, and a formatted path.
+func ConstructGetURL(
+	rawURL string, queries map[string][]string, format string, values ...interface{}) (*url.URL, error) {
+
+	urlPath, err := url.Parse(rawURL)
+	if err != nil {
+		return nil, err
+	}
+
+	urlPath.Path = fmt.Sprintf(format, values...)
+	var q url.Values = queries
+	urlPath.RawQuery = q.Encode()
+
+	return urlPath, nil
+}
+
+// GetAndParseJSON invokes a HTTP request with basic auth.
+// On success the response body will be unmarshalled as JSON.
+func GetAndParseJSON(result interface{}, user, token string, urlPath *url.URL) error {
+	body, err := getBodyFromRequest(user, token, urlPath)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(body, &result)
+
+	return err
+}
+
+// GetAsString invokes a HTTP request with basic auth.
+// Returns the response as a string.
+func GetAsString(user, token string, urlPath *url.URL) (string, error) {
+	body, err := getBodyFromRequest(user, token, urlPath)
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func getBodyFromRequest(user string, token string, urlPath *url.URL) ([]byte, error) {
+	url := urlPath.String()
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return []byte{}, err
+	}
```

Comment:
Think a good alternative would be to create a new type, for example named `azureClient`, that you feed in the credentials once instead of having to add them in every request function

Reply:
In 8d3c659 I've extracted most things from azuredevops.go into `importer.AzureDevOpsImporter`, for lack of a better name.

What do you reckon is the main task of an "importer" vs. the main task of a "client"? I'm having some trouble thinking of a good distinction between the two.

Reply:
Client: basic functions to talk to the Azure DevOps APIs. Such as GetFile, ListProjects, et.al.

Importer: Integration between the Wharf client and Azure DevOps client. It uses the basic functions from the clients to coordinate the data transfer.

Just to separate the responsabilities and de-bloat the code

Reply:
I've done what I imagine you meant in 1e1c3d4, but it feels pretty messy.
If you have any suggestions on how to make it cleaner, I'm all ears!